common/rpc/encryption/localStoreTlsFactory.go

```diff
@@ -154,13 +154,11 @@ func newServerTLSConfig(
 	certProvider CertProvider,
 	perHostCertProviderFactory PerHostCertProviderFactory,
 ) (*tls.Config, error) {
-	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider)
-	if err != nil {
-		return nil, err
-	}
 
-	if tlsConfig != nil && perHostCertProviderFactory != nil {
-		tlsConfig.GetConfigForClient = func(c *tls.ClientHelloInfo) (*tls.Config, error) {
+	var callback func(c *tls.ClientHelloInfo) (*tls.Config, error)
+
+	if perHostCertProviderFactory != nil {
```

Comment:
[Nit] You could just put the main logic of this if/else block into a single callback and directly assign it. e.g.

tlsConfig.GetConfigForClient := func(c *tls.ClientHelloInfo) (*tls.Config, error) {
    if perHostCertProviderFactory != nil {
       ...
       if perHostCertProvider != nil {
            return getServerTLSConfigFromCertProvider(perHostCertProvider)
       }
    }

    return getServerTLSConfigFromCertProvider(certProvider)
}